search.py

```diff
@@ -32,11 +35,26 @@ def to_list(self, amount, threshold):
         return items if len(items) > 0 else None
 
 
+def cached_parsing(func):
+    @functools.wraps(func)
+    def checking_cache_time(self, *args, **kwargs):
+        if dtm.date.today() > self.cache_time:
+            self._parse()
+            self.cache_time = dtm.date.today()
+        return func(self, *args, **kwargs)
+    return checking_cache_time
+
+
 class Search:
     def __init__(self):
         self._docs = {}
         self._official = {}
         self._wiki = OrderedDict()  # also examples
+        self.cache_time = dtm.date.today()
```

Comment:
Shouldn't this be `self.cache_date` since you're not storing the time at all? Could also make it even clearer and call it `self.last_cache_date` but mweh.